src/implementations/twig/components/form-group/form-group.html.twig

```diff
@@ -58,35 +65,62 @@
     disabled: true,
   }) %}
 {% endif %}
+
 {% if _invalid %}
   {% set _input = _input|merge({
     invalid: true,
   }) %}
+  {% if _id %}
+    {% set _aria_describedby = _aria_describedby ? _aria_describedby ~ ' ' ~ _id ~ '-invalid' : _id ~ '-invalid' %}
+  {% endif %}
+  {% set _input_extra_attributes = _input_extra_attributes|merge([
+    { name: 'aria-invalid', value: 'true' }
+  ]) %}
 {% endif %}
+
 {% if _required %}
   {% set _input = _input|merge({
     required: true,
   }) %}
 {% endif %}
 
 {% if _input.input_type == 'range' %}
-  {% set extra_attributes = [
+  {% set extra_attributes = extra_attributes|default([])|merge([
     { name: 'data-ecl-range', value: true },
     { name: 'data-ecl-auto-init', value: 'Range' },
-  ] %}
-{% endif %}
-
-{% if _input.input_type == 'rating-field' and _helper_id is not empty %}
- {% set extra_attributes = [
-  { name: 'aria-describedby', value: _helper_id }
- ] %}
+  ]) %}
 {% endif %}
 
 {% if _input.input_type == 'file' %}
-  {% set extra_attributes = [
+  {% set extra_attributes = extra_attributes|default([])|merge([
     { name: 'data-ecl-file-upload-group', value: true },
     { name: 'data-ecl-autoinit', value: 'FileUpload' },
-  ] %}
+  ]) %}
+{% endif %}
+
+{% if _helper_text is not empty %}
+  {% set _input_extra_attributes = _input_extra_attributes|merge([
+    { name: 'aria-describedby', value: _aria_describedby }
+  ]) %}
+{% endif %}
+
+{% if _input.input_type == 'radio' or _input.input_type == 'checkbox' or _input.input_type == 'rating-field' %}
+  {% set _group_tag = 'fieldset' %}
+  {% set _label_tag = 'legend' %}
+  {% if _aria_describedby %}
+    {% set extra_attributes = extra_attributes|default([])|merge([
+      { name: 'aria-describedby', value: _aria_describedby },
+    ]) %}
+  {% endif %}
+  {% if _invalid %}
+    {% set extra_attributes = extra_attributes|default([])|merge([
+      { name: 'aria-invalid', value: 'true' },
+    ]) %}
+  {% endif %}
+{% elseif _input_extra_attributes is not empty %}
```

Comment:
i'm terribly confused by this logic about the input.extra_attributes, this merge here only happens for input_type different than radio, checkbox rating field, but we do set some value for them all when are invalid, for instance.
Most likely this is done on purpose, and these attributes are then set in the input, probably because of some differences in the markup of these input types, if that is the case maybe we could add some comment in the template to explain what happens.

Reply:
There are two things here: for some input type (checkbox, radio and rating), the extra attribute has to be added to the wrapper element, using `extra_attributes`. 
For the other cases, it is added on the input element, using `input.extra_attributes`.
I'll add some comments